avocado/utils/genio.py

```diff
@@ -191,3 +193,35 @@ def is_pattern_in_file(filename,  pattern):
         if re.search(pattern, content_file.read(), re.MULTILINE):
             return True
     return False
+
+
+def get_file_hash(filename):
```

Comment:
This looks *a lot* like `avocado.utils.crypto.hash_file`.  While yours has the advantage that it uses `io.DEFAULT_BUFFER_SIZE` (very good!), `avocado.utils.crypto.hash_file` allows a one to (optionally) choose the hash algorithm.  Because everything on `avocado.utils` is intended not only for our own use, but for test writers, we should consider improving/reusing existing utilities.

Reply:
I was looking for the existing solution but I miss crypto.hash_file, thanks, I will use that. And I also created PR #3262 for `io.DEFAULT_BUFFER_SIZE` inside `avocado.utils.crypto.hash_file`